notebooks/week5/01.preprocess.py

```
@@ -0,0 +1,94 @@
+1. Load source dataset and retrieve recent records with updated timestamps.
+2. Split new records into train and test sets (80-20 split).
+3. Append new train and test records to existing train and test tables.
+4. Insert the latest feature data into the feature table for online serving.
+5. Trigger a pipeline update and monitor its status until completion.
+6. Set a task value indicating whether new data was processed.
+"""
+
+import argparse
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as F
+import yaml
```

Ruff

`yaml` imported but unused

Remove unused import: `yaml`

(F401)